controllers/tests/controller_server/host_definer/storage_class_watcher_test.py

```diff
@@ -0,0 +1,82 @@
+            test_utils.get_fake_k8s_storage_class_items(settings.CSI_PROVISIONER_NAME)
+        self.storage_class_watcher.host_definitions_api.get.return_value = \
+            test_utils.get_fake_k8s_host_definitions_items(settings.READY_PHASE)
+        self.storage_class_watcher.core_api.read_namespaced_secret.return_value = test_utils.get_fake_k8s_secret()
+
+    def test_add_new_storage_class_with_new_secret(self):
+        self.storage_class_watcher.host_definitions_api.get.return_value = \
+            test_utils.get_fake_k8s_host_definitions_items('not_ready')
+        self.nodes_on_watcher_helper[settings.FAKE_NODE_NAME] = settings.FAKE_NODE_ID
+        self.storage_class_watcher.add_initial_storage_classes()
+        self.storage_class_watcher.storage_host_servicer.define_host.assert_called()
+        self.assertEqual(1, len(self.secret_ids_on_storage_class_watcher))
+
+    def test_add_new_storage_class_with_existing_secret(self):
+        self.secret_ids_on_storage_class_watcher[settings.FAKE_SECRET_ID] = 1
+        self.storage_class_watcher.add_initial_storage_classes()
+        self.storage_class_watcher.storage_host_servicer.define_host.assert_not_called()
+        self.assertEqual(2, self.secret_ids_on_storage_class_watcher[settings.FAKE_SECRET_ID])
+
+    def test_add_new_storage_class_without_ibm_csi_provisioner(self):
+        self.storage_class_watcher.storage_api.list_storage_class.return_value = \
+            test_utils.get_fake_k8s_storage_class_items(settings.FAKE_CSI_PROVISIONER)
+        self.storage_class_watcher.add_initial_storage_classes()
+        self.storage_class_watcher.storage_host_servicer.define_host.assert_not_called()
+
+
+class TestWatchStorageClassResources(StorageClassWatcherBase):
+    def setUp(self):
+        super().setUp()
+        self.storage_class_stream = patch('{}.watch.Watch.stream'.format(settings.SECRET_WATCHER_PATH)).start()
+        self.storage_class_stream.return_value = iter([test_utils.get_fake_secret_storage_event(
+            settings.ADDED_EVENT, settings.CSI_PROVISIONER_NAME)])
+        self.storage_class_watcher.host_definitions_api.get.return_value = \
+            test_utils.get_fake_k8s_host_definitions_items(settings.READY_PHASE)
+        self.storage_class_watcher.core_api.read_namespaced_secret.return_value = test_utils.get_fake_k8s_secret()
+        self.storage_class_watcher._loop_forever = Mock()
+        self.storage_class_watcher._loop_forever.side_effect = [True, False]
+
+    def test_add_new_storage_class_with_new_secret(self):
+        self.storage_class_watcher.host_definitions_api.get.return_value = \
+            test_utils.get_fake_k8s_host_definitions_items('not_ready')
+        self.nodes_on_watcher_helper[settings.FAKE_NODE_NAME] = settings.FAKE_NODE_ID
+        self.storage_class_watcher.watch_storage_class_resources()
+        self.storage_class_watcher.storage_host_servicer.define_host.assert_called()
```

Comment:
can we be more specific with `assert_called`?

Reply:
what do you mean by being more specific?

Reply:
i.e. assert_called_once_with(field="value")

Reply:
done